src/java.base/share/classes/com/sun/crypto/provider/HmacMD5KeyGenerator.java

```diff
@@ -88,6 +88,11 @@ protected void engineInit(AlgorithmParameterSpec params,
      * @param random the source of randomness for this key generator
      */
     protected void engineInit(int keysize, SecureRandom random) {
+
+        if (keysize <= 0) {
+            throw new IllegalArgumentException("keysize must not be <= 0 for SunJCE");
```

Comment:
< 80 chars please

Reply:
The "for SunJCE" is odd to see in an exception message, I would leave that out.

Reply:
A stack trace of the exception should be sufficient to tell what provider/code is generating it.

Reply:
@seanjmullan I thought about this, but I added it due to the bug describing that this would be implementation-specific rather than framework-specific.

Reply:
> Needs regression tests.
> 
> Have you looked at the other Key(Pair)Generators? We've probably added more when this bug was filed. I took a quick look at some, and they were covered. (DES/DESede/Blowfish/AES/etc.)

@bradfordwetmore The bug mentions only the Hmac*KeyGenerators. I thought about this too, but you mentioned in your initial triage comment on the bug that it should be treated as only relevant to the classes I modified. Thoughts?

Reply:
What bug/Sean meant was that we wouldn't put in a check into java.security.* framework, but rather in sun.security.* implementation.  And that we don't need to put "SunJCE" in this particular throws message, as we'll know where it is from the stack trace.

My comment agrees that we have a problem in the JCE Hmac code, but there might be other locations where init's like this might allow for invalid 0 values.  It's just a general idea to check we didn't make the same issue elsewhere.

Reply:
Other (newer) HmacXXX KeyGenerator enforces min key size to be 40. So, perhaps just enforce the same lower limit or just check for <0?

Reply:
Also, it'd be nice to include HmacMD5 and HmacSHA1 to the bug synopsis as KeyGenerator covers many algorithms. Would be nice to add SunJCE provider to synoposis, so it's clear that this change/fix is provider-specific.